db/src/tx.rs

```diff
@@ -130,6 +130,11 @@ use watcher::{
     TransactWatcher,
 };
 
+pub(crate) enum TransactorAction {
```

Comment:
This is where the nice comment explaining what is happening in the two cases goes :)